File: Python_Programming/Day_3/program15.py
# Description : Write a program to check whether the number is perfect number or not.

def Factorial(No):
    Sum = 0
    
    for i in range(1 , (No//2)+1 , 1):
        if(No % i == 0):
            Sum = Sum + i

    return (Sum == No)

def main():

    Value = int(input("Enter the number : "))

    Ret = Factorial(Value)

    if(Ret == True):
        print("It's an perfect number...")
    else:
        print("It's not an perfect number...")

if __name__ == "__main__":
    main()

File: C_Programming/Day_3/program15.c
// Description : Write a program to check whether the number is perfect number or not.

# include<stdio.h>
#include<stdbool.h>

bool CheckPerfect(int iNo)
{
    int iCnt = 0;
    int iSum = 0;

    for(iCnt = 1 ; iCnt <= iNo/2 ; iCnt++)
    {
        if(iNo % iCnt == 0)
        {
            iSum = iSum + iCnt;
        }
    }

    return (iSum == iNo);
}


int main()
{
    int iValue = 0;
    bool bRet = false;

    printf("Enter the number : ");
    scanf("%d",&iValue);

    bRet = CheckPerfect(iValue);

    if(bRet == true)
    {
        printf("It is a perfect number...\n");
    }
    else
    {
        printf("It is not a perfect number...\n");

    }
    return 0;
}

